### ISSUE TITLE
The type initializer for 'LanguageExt.ClassInstances.OrdDefault`1' threw an exception.

### ISSUE THREAD

**Issue by vitalmiguel** [louthy/language-ext#753]
Hello,

I've been trying to create a Map<Enum, string> but I'm always getting this exception: 

`The type initializer for 'LanguageExt.ClassInstances.OrdDefault`1' threw an exception.`

Here's the stacktrace:
```
at LanguageExt.ClassInstances.OrdDefault`1.Compare(A x, A y)
at LanguageExt.MapModuleM.Add[OrdK,K,V](MapItem`2 node, K key, V value, AddOpt option)
at LanguageExt.MapInternal`3..ctor(IEnumerable`1 items, AddOpt option)
at LanguageExt.MapInternal`3.TryAddRange(IEnumerable`1 range)
at LanguageExt.Map`2.TryAddRange(IEnumerable`1 range) 
```

I've tried initializing Map with KeyValuePairs, Tuple, each are some of the overloads that the ctor of Map supports but without success.
I can initialize as empty Map, I can also use Add(enum, string) but if I try to AddRange(), use toMap() I always get the same error.

Am I doing something wrong? I also tried to update to the lastest version of the package but I'm getting this exception anyway.

Here's some code I've tried:
```csharp
var somes = seq.Somes();
var kvs = somes.Map(o => new KeyValuePair<StateEnum, string>(o.Type, o.Value));
try
{
     var result = toMapTry(kvs);
}
catch (Exception ex)
{
     string asa = null;
}
```

**Comment by Shammah** [comment]
I've had the same issue with the latest stable version (3.4.15). I had one library project that created a map, and my executable project using that map, which gave me the exception. When I created the map directly in my executable project, without using the library project, I did not have the issue.

However, I've tried upgrading to the latest beta release (3.5.4-beta) and the problem seems to be fixed for me this way.
